CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -282,24 +282,26 @@ public static function getfinancialAccountRelations($flip = FALSE) {
     $params = array('labelColumn' => 'name');
     $financialAccountType = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_FinancialAccount', 'financial_account_type_id', $params);
     $accountRelationships = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_EntityFinancialAccount', 'account_relationship', $params);
-    $Links = array(
-      'Expense Account is' => 'Expenses',
-      'Accounts Receivable Account is' => 'Asset',
-      'Income Account is' => 'Revenue',
-      'Asset Account is' => 'Asset',
-      'Cost of Sales Account is' => 'Cost of Sales',
-      'Premiums Inventory Account is' => 'Asset',
-      'Discounts Account is' => 'Revenue',
-      'Sales Tax Account is' => 'Liability',
-      'Deferred Revenue Account is' => 'Liability',
-    );
+    if (!isset(Civi::$statics[__CLASS__]['account_relationships'])) {
```

Comment:
@monishdeb I'm confused now - aren't we just caching a static array here?

Reply:
Oh I see - it's a 'pseudohook' - we probably should do an actual test & add a unit test if we actually support people altering this array

Reply:
Or perhaps given "Although the list of available account relationships lived in the database, the labels for each relationship were hardcoded in code in a non-hookable way." - se should retrieve it from the databse!

Reply:
Hmm issue is that FA relationship type like 'Expenses', 'Revenue' etc can't be fetched against FA relationship types from DB, reason why I relied on in this hardcoded format and lets user manipulate this mapping list via an extension.

Reply:
Although civicrm_entity_financial_account does store the mapping for some but not all, but that too if any FT has assigned FAs

Reply:
I can write a unit test to show that we can manipulate the array inside a hook and later check the final list by calling this function directly. What do you think?

Reply:
It is possible to extend the mapping from the current technique too but lets dev. be aware of this extendability feature for which I would prefer new hook not separate option value set.  Because the mapping is only used in this fn and at FA form level. So I won't put burden in option set as its usage is not much in codebase :)

Reply:
Well I guess so but I think adding a hook might be as big a 'burden' in that we don't normally have a hook for a single set of configurable values - maybe @totten or @colemanw have a suggestion